src/main/java/org/apache/cassandra/sidecar/data/StreamSSTableComponentRequest.java

```diff
@@ -43,24 +45,74 @@ public class StreamSSTableComponentRequest extends SSTableComponent
     @VisibleForTesting
     public StreamSSTableComponentRequest(String keyspace, String tableName, String snapshotName, String componentName)
     {
-        this(new QualifiedTableName(keyspace, tableName, true), snapshotName, componentName);
+        this(new QualifiedTableName(keyspace, tableName, true), snapshotName, null, componentName);
+    }
+
+    /**
+     * Constructor for the holder class
+     *
+     * @param keyspace      the keyspace in Cassandra
+     * @param tableName     the table name in Cassandra
+     * @param snapshotName  the name of the snapshot
+     * @param indexName     the name of the index for the SSTable component
+     * @param componentName the name of the SSTable component
+     */
+    @VisibleForTesting
+    public StreamSSTableComponentRequest(String keyspace, String tableName, String snapshotName, String indexName,
+                                         String componentName)
+    {
+        this(new QualifiedTableName(keyspace, tableName, true), snapshotName, indexName, componentName);
+    }
+
+    /**
+     * Constructor for the holder class
+     *
+     * @param qualifiedTableName the qualified table name in Cassandra
+     * @param snapshotName       the name of the snapshot
+     * @param componentName      the name of the SSTable component
+     */
+    public StreamSSTableComponentRequest(QualifiedTableName qualifiedTableName,
+                                         String snapshotName,
+                                         String componentName)
+    {
+        this(qualifiedTableName, snapshotName, null, componentName);
```

Comment:
Similar to the above comment. How about just modify the constructor? The class is for server internal usage, not a public API.